### PR TITLE
Bug 1310599 - Persist file fields on form submission

### PR DESCRIPTION
**Pull request by jezdez**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/telemetry-analysis-service/45)

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by jezdez** [comment]
@maurodoglio r?

File: Makefile [hunk 1]
```diff
@@ -39,7 +43,7 @@ stop:
 	docker-compose stop
 
 test: static
-	docker-compose run web ./manage.py test
+	docker-compose run --service-ports web ./manage.py test
```
Comment (jezdez) [inline Makefile]: woops, that was just for debugging purposes, removing..

File: atmo/forms/mixins.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        """
+        super(CreatedByFormMixin, self).clean()
+        if self.instance.id and self.created_by != self.instance.created_by:
+            raise forms.ValidationError(
+                'Access denied to the data of another user'
+            )
+
+
+class CachedFileFormMixin(object):
+
```
Comment (jezdez) [inline atmo/forms/mixins.py]: 👍

File: atmo/forms/mixins.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        for name in self.cached_filefields:
+            self.cache.remove(self.cachekey_input_data(name))
+        return super(CachedFileFormMixin, self).save(*args, **kwargs)
+
+    def clean(self):
+        for field_name in self.cached_filefields:
+            # get the name of the cache key field name and its value
+            cache_key = self.cachekey_input_data(field_name)
+
+            # check form data if the file field if it was submitted
```
Comment (jezdez) [inline atmo/forms/mixins.py]: 👍

**Pull request merged**